### DISCUSSION TITLE
Any way to delete a lines content but not the line itself

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4047] [Q&A] [answered]
I apologize if I am overlooking this, but I was wondering if anyone knew if it was possible to be able to delete the contents of a line (or multiple lines) but still keep the existing line numbers present.

Thank You

**Comment by ntrel** [comment]
There are *Delete to line end* and *Delete to line start* keybindings:
https://www.geany.org/manual/current/index.html#editor-keybindings

**Comment by advice2020** [comment]
Thank you for your response.
This requires cursor to have to be at the beginning of the line to work correctly for what I am looking for.
Also that being said this does not work when selecting multiple lines, only works one line per time.
Looking for something more like Ctrl+K "Delete current line(s)" type of functionality, without the removing of the actual line (line number) itself.

**Comment by Skif-off** [comment]
Maybe GeanyLua and something like
```lua
local sel = geany.selection()
if (sel ~= nil) and (sel ~= "") then
  sel = string.gsub(sel, "[^\r\n]", "")
  geany.selection(sel)
end
```
?
But the lines must be selected in full.

If the lines are not fully selected or we need to clear one line under the cursor, we will have to use `geany.rowcol`, but this function behaves strangely with the second parameter. I haven't figured out what's going on yet.

**Comment by Skif-off** [comment]
This should work fine in any case
```lua
local start, stop
local n1, n2 = geany.select()
if n1 > n2 then
  start, stop = n2, n1
else
  start, stop = n1, n2
end
local line, column = geany.rowcol(start)
start = geany.scintilla("SCI_POSITIONFROMLINE", line - 1)
line, column = geany.rowcol(stop)
stop = geany.scintilla("SCI_GETLINEENDPOSITION", line - 1)
geany.select(start, stop)
local sel = geany.selection()
sel = string.gsub(sel, "[^\r\n]+", "")
geany.selection(sel)
if n1 > n2 then geany.select(start) end

--[[
local script = string.gsub(geany.script, "^[^\r\n]+" .. geany.dirsep .. "([^" .. geany.dirsep .. "]+)$", "%1")
geany.status(script .. ": Done!")
]]
```

**Comment by b4n** [comment, marked as answer]
I'm not sure what the use case can be, but you possibly could use the *send selection to* feature, with e.g. the command `sed 's/^.*$//'`. This requires minimal configuration and no plugin.

It'll work fine with either no selection (will handle the whole current line) or with full lines selected. If lines are partially selected it'll leave the unselected bits as they are.

**Comment by advice2020** [comment]
@b4n
Thank you for your response.
I had seen these Send to Custom Commands items in the keybindings and it was on my list of things to try,
your response gave me a reason to do so.
I tested your solution out, works great.
I have a couple use cases, but will just share one.
This can be helpful if you have scripts where you need to remove blobs from the middle and then have to replace that text.  This helps by not having to delete the blobs and have to hit Enter key a bunch of times to get the space back to type.  Also this space is helpful visually to indicate where the code was removed from.

Thank You

**Comment by advice2020** [comment]
@Skif_off
Wow, thank you for this my friend, this works great as well.
I have a couple of tweaks/questions if interested.
These might not be possible, because I do not know what Lua has access to in Geany.
Definitely keep the script you provided above as mentioned works great, I am going to use it.
-Would it be possible/easy to copy that script and make a version that applies a "select all” and then performs the content delete?

-Okay if this is not possible/not interested because I could just use the solution that "b4n" mentioned if I require this specific behavior, but I noticed that when using that “sed” solution provided, that when it performs the deleting of the lines content, that it advances the cursor to the next line automatically, so it can be used repeatedly, which in certain situations can be use as well.  Would it be possible to add this functionality to a separate copy of the script you provided?
I am just thinking, if I am going to use Lua scripts in Geany for a few things, nice to have them all use the same Geany plugin.

Thanks again

**Comment by advice2020** [comment]
Thank you both again, although it would be nice if this was an action provided by Geany itself, these are two great ideas for providing solutions for this.

**Reply by eht16** [reply]
@advice2020 would you mind marking one of the replies as answer?

**Comment by Skif-off** [comment]
> Would it be possible/easy to copy that script and make a version that applies a "select all” and then performs the content delete?

"Select all" sounds like an unnecessary action, let's just try to replace the text
```lua
local iLines = geany.height()
local iEOL = geany.scintilla("SCI_GETEOLMODE")
local sLE
if iEOL == 0 then
  sLE = "\r\n"
elseif iEOL == 1 then
  sLE = "\r"
elseif iEOL == 2 then
  sLE = "\n"
else
  return
end
local sTmp = ""
for i = 1, iLines - 1 do
  sTmp = sTmp .. sLE
end
geany.text(sTmp)
```

> Would it be possible to add this functionality to a separate copy of the script you provided?

Try to replace
```lua
if n1 > n2 then geany.select(start) end
```
on this
```lua
start = geany.rowcol(line + 1, 0)
geany.select(start)
```

**Comment by advice2020** [comment]
I do not mind marking one as the answer.
I will mark the solution provided by b4n as the answer because as he mentioned it does not require the use of a plugin, it uses a built in feature of Geany.

**Comment by advice2020** [comment]
@Skif_off
.....wow, the two new updates, along with the original script work great.  Yeah the “select all” was the easiest way to describe what I was looking to achieve, but your solution as mentioned works great.
Thank you so much for all of this.
